### ISSUE TITLE
Incorrect predictions 

### ISSUE THREAD

**Issue by raina0990** [osmr/imgclsmob#63]
I always get incorrect predictions for an image.

Script:
rom kerascv.model_provider import get_model as kecv_get_model
from keras.preprocessing import image
from keras.applications.imagenet_utils import preprocess_input,decode_predictions
import numpy as np
import tensorflow as tf
import json

  

f = 'limpkin.jpeg'
net = kecv_get_model("resnet18", pretrained=True)

img = image.load_img(f, target_size=(224,224))
transform_image = image.img_to_array(img)
out_image = np.expand_dims(transform_image,axis=0)

preprocess_img = preprocess_input(out_image)
        

y = net.predict(preprocess_img)

sorted_indices = []
sorted_probs = []
max_value = float("-inf")

for i in range(y.size) :
    if (y.item(i) > max_value):
        max_value = y.item(i)
        maxId = i

        
        sorted_indices.append(maxId)
        sorted_probs.append(max_value)
        

sorted_indices.reverse()
sorted_probs.reverse()

class_idx = json.load(open("imagenet_class_index (1).json"))
idx2label = [class_idx[str(k)][1] for k in range(len(class_idx))]
predictions = []
for ind in range(5):
    print(ind)
    print(sorted_indices[ind])
    print(idx2label[sorted_indices[ind]])
    print(sorted_probs[ind])
    predictions.append([idx2label[sorted_indices[ind]], sorted_probs[ind]])
---------------------------
Output:
Using TensorFlow backend.
WARNING:tensorflow:From /usr/local/lib/python3.7/site-packages/tensorflow_core/python/ops/resource_variable_ops.py:1630: calling BaseResourceVariable.__init__ (from tensorflow.python.ops.resource_variable_ops) with constraint is deprecated and will be removed in a future version.
Instructions for updating:
If using Keras pass *_constraint arguments to layers.
2020-04-23 22:45:46.174955: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2020-04-23 22:45:46.207436: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x7fb3bacd1290 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2020-04-23 22:45:46.207454: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
WARNING:tensorflow:From /usr/local/lib/python3.7/site-packages/keras/backend/tensorflow_backend.py:4070: The name tf.nn.max_pool is deprecated. Please use tf.nn.max_pool2d instead.

WARNING:tensorflow:From /usr/local/lib/python3.7/site-packages/keras/backend/tensorflow_backend.py:4074: The name tf.nn.avg_pool is deprecated. Please use tf.nn.avg_pool2d instead.

WARNING:tensorflow:From /usr/local/lib/python3.7/site-packages/keras/backend/tensorflow_backend.py:422: The name tf.global_variables is deprecated. Please use tf.compat.v1.global_variables instead.

chain_saw
44.54088592529297
cheetah
36.67350387573242
limpkin
29.855144500732422
sidewinder
24.926227569580078
triceratops
23.750385284423828

**Comment by osmr** [comment]
Most likely you do preprocessing for Caffe. It's wrong. 

I recommend you to use `tf2cv` package instead of `kerascv`. See an example of using classification models for TensorFlow 2.0: https://github.com/osmr/imgclsmob/blob/master/examples/demo_tf2.py.

**Comment by raina0990** [comment]
Ok ! let me try this package

**Comment by raina0990** [comment]
I used the models in the kerascv library. As suggested I was doing incorrect pre processing of the image. After correcting the preprocessing the models in the kerascv gave correct results.
